File: Lua/bee 1040.lua
local n1 = tonumber(io.read("*n")) * 2
local n2 = tonumber(io.read("*n")) * 3
local n3 = tonumber(io.read("*n")) * 4
local n4 = tonumber(io.read("*n"))

local media = (n1 + n2 + n3 + n4)/10
print(string.format("Media: %.1f", media))
if media >= 7 then
    print("Aluno aprovado.")
elseif media < 5 then
    print("Aluno reprovado.")
else
    local n5 = tonumber(io.read("*n"))
    print("Aluno em exame.")
    print("Nota do exame: " .. n5)
    local mediaFinal = (n5 + media)/2
    if mediaFinal >= 5 then
        print("Aluno aprovado.")
    else 
        print("Aluno reprovado.")
    end
    print(string.format("Media final: %.1f", mediaFinal))
end

File: Rust/bee_1040.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();

    let valores: Vec<f64> = input
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let n1 = valores[0] * 2.0;
    let n2 = valores[1] * 3.0;
    let n3 = valores[2] * 4.0;
    let n4 = valores[3];

    let media = (n1 + n2 + n3 + n4) / 10.0;

    println!("Media: {:.1}", media);

    if media >= 7.0 {
        println!("Aluno aprovado.");
    } else if media < 5.0 {
        println!("Aluno reprovado.");
    } else {
        println!("Aluno em exame.");

        let n5 = valores[4];
        println!("Nota do exame: {:.1}", n5);

        let media_final = (media + n5) / 2.0;
        println!(
            "{}",
            (if media_final >= 5.0 {
                "Aluno aprovado."
            } else {
                "Aluno reprovado."
            })
        );

        println!("Media final: {:.1}", media_final);
    }
}
